akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -458,9 +457,11 @@ private[akka] class ActorCell(
   private[this] var sysmsgStash: LatestFirstSystemMessageList = SystemMessageList.LNil
 
   // Java API
-  final def getParent() = parent
+  final override def getParent(): ActorRef = parent
```

Comment:
Hmm, so it was `InternalActorRef` before. While I like the change, I'm not sure this is safe: could there be a `ClassCastException` lurking if we change it? Of course this is less likely with Java because of the lack of type inference, but still seems scary.

Reply:
I agree, this shouldn't be visible to end users so better to just leave it as it was. The visible API is in ActorContext.